Sources/Citadel/Shell/EchoShell/BasicCommands.swift

```diff
@@ -0,0 +1,101 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Citadel open source and the ClamShell project
+//
+// Copyright (c) 2023 Gregor Feigel and the Citadel project authors
+// Licensed under MIT License
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: MIT
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+import NIO
+import NIOSSH
+
+// Basic Commands
+// Each command must return an array of UINT8 or nil. Streams are currently not supported.
+
+public protocol Command {
+    var name: String { get }
+    var description: String { get }
+    func exec(input: [UInt8], session: EchoShellMaster) async throws -> [UInt8]?
```

Comment:
I'm fine with changing the name to `CommandContext` and offering `CommandInput` with the ByteBuffer.
I personally tend to avoid working with strings at cmd-line level as basically everything is within the UTF-8/ASCI range and `String` just adds a lot of unnecessary overhead.

Reply:
I'm happy with changing the return to an `AsyncStream `.